tests/test_xyz/test_points.py

```diff
@@ -0,0 +1,135 @@
+import itertools
+import pathlib
+from hypothesis import given, strategies as st
+
+import pandas as pd
+import numpy as np
+import pytest
+
+from xtgeo.xyz import Points
+import tests.test_common.test_xtg as tsetup
+
+PFILE = pathlib.Path("points/eme/1/emerald_10_random.poi")
+POINTSET2 = pathlib.Path("points/reek/1/pointset2.poi")
+POINTSET3 = pathlib.Path("points/battle/1/many.rmsattr")
+POINTSET4 = pathlib.Path("points/reek/1/poi_attr.rmsattr")
+CSV1 = pathlib.Path("3dgrids/etc/gridqc1_rms_cellcenter.csv")
+
+
+def test_custom_points():
+    """Make points from list of tuples."""
+
+    plist = [(234, 556, 11), (235, 559, 14), (255, 577, 12)]
+
+    mypoints = Points(plist)
+
+    x0 = mypoints.dataframe["X_UTME"].values[0]
+    z2 = mypoints.dataframe["Z_TVDSS"].values[2]
+    assert x0 == 234
+    assert z2 == 12
+
+
+@st.composite
+def list_of_equal_length_lists(draw):
+    list_len = draw(st.integers(min_value=3, max_value=4))
+    fixed_len_list = st.lists(
+        st.floats(allow_nan=False, allow_infinity=False),
+        min_size=list_len,
+        max_size=list_len,
+    )
+    return draw(st.lists(fixed_len_list, min_size=1))
+
+
+@given(list_of_equal_length_lists())
+def test_create_pointset(points):
+    pointset = Points(points)
+    points = np.array(points)
+
+    assert len(points) == pointset.nrow
+
+    np.testing.assert_array_almost_equal(pointset.dataframe["X_UTME"], points[:, 0])
+    np.testing.assert_array_almost_equal(pointset.dataframe["Y_UTMN"], points[:, 1])
+    np.testing.assert_array_almost_equal(pointset.dataframe["Z_TVDSS"], points[:, 2])
+
+
+def test_import(testpath):
+    """Import XYZ points from file."""
+
+    mypoints = Points(testpath / PFILE)  # should guess based on extesion
+
+    x0 = mypoints.dataframe["X_UTME"].values[0]
+    tsetup.assert_almostequal(x0, 460842.434326, 0.001)
+
+
+def test_import_from_dataframe(testpath):
+    """Import Points via Pandas dataframe"""
```

Comment:
Also docstrings shall end with periods (sorry for being picky)

Reply:
Will fix! :)

Reply:
Just to avoid any confusion, the commit `CLN: re-arrange tests` which some of the docstring comments refer to are only rearranging the tests, I am not editing any of it.